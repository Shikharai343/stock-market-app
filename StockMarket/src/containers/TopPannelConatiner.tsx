import React, { Component } from 'react';
import { FlatList, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {RouteNames} from '../navigation/routes';
import Airtable from 'airtable';
import axios from 'axios';
import {NavigationActions, StackActions} from 'react-navigation';

interface IState {
    dataSource: object;
    stockData: object[];
}

interface IProps {
    navigation: {
        navigate: (route: string, data: object) => void;
        dispatch: any;
    };
}

class TopPannelConatiner extends Component<IProps, IState> {
    public state = {
        dataSource: {},
        stockData: [],
    };

    public componentDidMount() {
        axios.get('https://api.airtable.com/v0/appaf01a2JbZoZNee/stock%20market?api_key=keyiMCbcYlCf5VXsP')
            .then((res) => {
                this.setState({
                    stockData: res.data.records,
                });

            });

        const items = Array.apply(null, Array(30)).map((v, i) => {
            return (i + 1);
        });
        this.setState({
            dataSource: items,
        });
    }

    public handlePress = (item: object, itemPrice?: undefined | string, updateState?: any, deleteData?: any) => {
        this.props.navigation.navigate(RouteNames.stock, {
            item,
            itemPrice,
            updateState,
            deleteData,
        });
    };

    public render() {
       return (
            <View style={styles.container}>
            <FlatList
                data={this.state.dataSource}
                renderItem={this.renderListItem}
                ListHeaderComponent={this.renderHeader}
                numColumns={3}
                keyExtractor={this.keyExtractor}
                extraData={this.state.stockData.length}
            />
            </View>
        );
    }

    private renderListItem = ({item}: {item: any}): React.ReactElement => {
        const itemPrice = this.state.stockData
                                    .find((input: any) => item === new Date(input.fields.Date).getDate());

        const updateState = (text: any) => {
            const base = new Airtable({apiKey: 'keyiMCbcYlCf5VXsP'}).base('appaf01a2JbZoZNee');

            if(itemPrice !== undefined){
                base('stock market').update(itemPrice.id, {
                    "Close": text
                }, function(err: any, record: any) {
                    if (err) {
                        console.error(err);
                        return;
                    }
                });
            }else {
                base('stock market').create({
                    "Date": `2019-06-${item}`,
                    "Close": text,
                }, function(err: any, record: any) {
                    if (err) {
                        console.error(err);
                        return;
                    }
                });
            }

            const resetAction = StackActions.reset({
                index: 0,
                actions: [NavigationActions.navigate({ routeName: 'Home' })],
            });
            this.props.navigation.dispatch(resetAction);
        };

        const deleteData = () => {
            var base = new Airtable({apiKey: 'keyiMCbcYlCf5VXsP'}).base('appaf01a2JbZoZNee');

            if(itemPrice !== undefined) {
                base('stock market').destroy(itemPrice.id, function(err: any, deletedRecord: any) {
                    if (err) {
                        console.error(err);
                        return;
                    }
                });
            }

            const resetAction = StackActions.reset({
                index: 0,
                actions: [NavigationActions.navigate({ routeName: 'Home' })],
            });
            this.props.navigation.dispatch(resetAction);
        };

        const handleItemPress = () => this.handlePress(
            item,
            itemPrice && itemPrice.fields.Close,
            updateState, deleteData
        );

        return (
            <TouchableOpacity style={styles.dateContainer} onPress={handleItemPress}>
                <View style={styles.box}>
                    <Text style={styles.itemStyle}>
                        {item}
                    </Text>
                    {
                        itemPrice &&
                        <Text>
                            ${itemPrice.fields.Close}
                        </Text>
                    }
                </View>
            </TouchableOpacity>
        );
    }

    private renderHeader = () => {
        return (
            <View style={styles.header}>
                <Text style={styles.textStyle}>
                    JUNE 2019
                </Text>
            </View>
        );
    }

    private keyExtractor = (item: any, index: number) => index.toString();
}

const styles = StyleSheet.create({
    container: {
        justifyContent: 'center',
        flex: 1,
    },
    imageThumbnail: {
        justifyContent: 'center',
        alignItems: 'center',
        height: 80,
    },
    dateContainer: {
        flex: 1,
        flexDirection: 'column',
        margin: 1,
    },
    header: {
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        height: 70,
        marginBottom: 5,
        backgroundColor: '#5b73ba'
    },
    textStyle: {
        fontFamily: 'Georgia-Bold',
        color: 'white',
        fontSize: 25,
        letterSpacing: 0.13,
    },
    box: {
        alignItems: 'center',
        justifyContent: 'center',
        backgroundColor: '#d2d4d2',
        height: 70,
        margin: 1,
    },
    itemStyle: {
        fontSize: 20,
    },
});

export default TopPannelConatiner;
