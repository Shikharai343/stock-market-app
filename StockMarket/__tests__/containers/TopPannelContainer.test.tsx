import React from 'react';
import renderer from 'react-test-renderer';
import {shallow, ShallowWrapper} from 'enzyme';
import TopPannelConatiner from '../../src/containers/TopPannelConatiner';

const createTestProps = (props: Object) => ({
    ...props
});
let wrapper: any;
let props: any;

describe('BottomPannelContainer testing', () => {
    props = createTestProps({});
    wrapper = shallow(<TopPannelConatiner {...props}/>);

    it('should render snapshot', () => {
        const tree = renderer.create(wrapper).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
