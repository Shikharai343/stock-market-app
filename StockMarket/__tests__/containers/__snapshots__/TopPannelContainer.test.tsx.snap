// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomPannelContainer testing should render snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <RCTScrollView
    ListHeaderComponent={[Function]}
    data={
      Array [
        1,
        2,
        3,
        4,
        5,
        6,
        7,
        8,
        9,
        10,
        11,
        12,
        13,
        14,
        15,
        16,
        17,
        18,
        19,
        20,
        21,
        22,
        23,
        24,
        25,
        26,
        27,
        28,
        29,
        30,
      ]
    }
    disableVirtualization={false}
    extraData={0}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={3}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#5b73ba",
              "flexDirection": "row",
              "height": 70,
              "justifyContent": "center",
              "marginBottom": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Georgia-Bold",
                "fontSize": 25,
                "letterSpacing": 0.13,
              }
            }
          >
            JUNE 2019
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                1
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                2
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                3
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                4
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                5
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                6
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                7
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                8
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                9
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                10
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                11
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                12
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                13
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                14
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                15
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                16
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                17
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                18
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                19
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                20
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                21
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                22
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                23
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                24
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                25
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                26
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                27
              </Text>
            </View>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                28
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                29
              </Text>
            </View>
          </View>
          <View
            accessible={true}
            clickable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "margin": 1,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#d2d4d2",
                  "height": 70,
                  "justifyContent": "center",
                  "margin": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "fontSize": 20,
                  }
                }
              >
                30
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
